programs/coherence-beamsplitter/src/context.rs

```diff
@@ -46,17 +81,244 @@ pub struct RegisterToken<'info> {
+    pub orderer_etf_ata: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        seeds = [
+            b"Beamsplitter".as_ref(),
+        ],
+        bump = beamsplitter.bump,
+    )]
+    pub beamsplitter: Box<Account<'info, Beamsplitter>>,
+
+    // ========================= PROGRAMS ===========================================
+    /// The [Beamsplitter] [Account] that holds all of the Program's funds
+    pub rent: Sysvar<'info, Rent>,
+
+    pub associated_token_program: Program<'info, AssociatedToken>,
+
+    pub token_program: Program<'info, Token>,
     /// The [System] program.
     pub system_program: Program<'info, System>,
 }
+
+#[derive(Accounts)]
+pub struct Cohere<'info> {
+    pub prism_etf_mint: Account<'info, Mint>,
+
+    /// The Prism ETF [Account] that describes the assets being purchased
+    #[account(seeds = [b"PrismEtf".as_ref(), &prism_etf_mint.key().to_bytes(), &beamsplitter.key().to_bytes()], bump = prism_etf.bump, has_one = weighted_tokens)]
+    pub prism_etf: Box<Account<'info, PrismEtf>>,
+
+    #[account(seeds = [b"OrderState".as_ref(), &prism_etf_mint.key().to_bytes(), &orderer.key().to_bytes(), &beamsplitter.key().to_bytes()], bump =  order_state.bump, has_one = transferred_tokens)]
+    pub order_state: Box<Account<'info, OrderState>>,
+
+    pub weighted_tokens: AccountLoader<'info, WeightedTokens>,
+
+    #[account(mut)]
+    pub transferred_tokens: AccountLoader<'info, TransferredTokens>,
+
+    /// The [Signer] of the tx and owner of the [Deposit] [Account]
+    pub orderer: Signer<'info>,
+
+    /// The authority on the transfer mint
+    pub transfer_authority: AccountInfo<'info>,
+
+    /// The mint of the asset being transfered
+    pub transfer_mint: Account<'info, Mint>,
+
+    /// The [TokenAccount] that transfers out tokens
+    #[account(associated_token::mint = transfer_mint, associated_token::authority = orderer, mut)]
+    pub orderer_transfer_ata: Box<Account<'info, TokenAccount>>,
+
+    /// The [TokenAccount] that transfers in tokens
+    #[account(init_if_needed, associated_token::mint = transfer_mint, associated_token::authority = beamsplitter, payer = orderer)]
+    pub beamsplitter_transfer_ata: Box<Account<'info, TokenAccount>>,
+
+    /// The [Beamsplitter] [Account] that holds all of the Program's funds
+    #[account(
+        seeds = [
+            b"Beamsplitter".as_ref(),
+        ],
+        bump = beamsplitter.bump,
+    )]
+    pub beamsplitter: Box<Account<'info, Beamsplitter>>,
+
+    // ========================= PROGRAMS ===========================================
+    pub rent: Sysvar<'info, Rent>,
+
+    pub associated_token_program: Program<'info, AssociatedToken>,
+
+    pub token_program: Program<'info, Token>,
+
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct Decohere<'info> {
+    pub prism_etf_mint: Account<'info, Mint>,
+
+    /// The Prism ETF [Account] that describes the assets being purchased
```

Comment:
Is this comment still correct? In Decohere we’re disuniting the ETF, right? So are we purchasing the assets here? If we’re disuniting an ETF then the underlying assets should already be owned. Is it correct to say this account describes the assets being purchased?